### ISSUE TITLE
Support new functions of GPGME 1.7

### ISSUE THREAD

**Issue by duckdalbe** [ueno/ruby-gpgme#84]
Do you plan to work on support the new functions of GPGME 1.7 soon?

I'm especially interested in gpgme_op_createkey, gpgme_op_createsubkey, gpgme_op_adduid, and gpgme_op_revuid. They would make my life easier. Unfortunately I'm not fluent enough in C to do it myself.

**Comment by ueno** [comment]
Yes.  When updating the bundled GPGME tarball we usually expose new functions, like https://github.com/ueno/ruby-gpgme/commit/46533d6de0d10d0b4d4b01995a949219815279a8.
PRs are welcome.

**Comment by duckdalbe** [comment]
This appears to have stalled?

Isn't there anyone interested in the new features and fluent enough in C to implement them?

**Comment by dansketcher** [comment]
Hey @duckdalbe - it's a pretty big question there. Is there a specific function(s) that you're in need of?
No point doing work no-one needs, but if there's demand that's a good place to start.

**Comment by duckdalbe** [comment]
Hey @dansketcher, thank you for your reply!

I'm mostly interested in the functions to create and manipulate keys.

* `gpgme_op_createkey`
* `gpgme_op_createsubkey`
* `gpgme_op_adduid`
* `gpgme_op_set_uid_flag`
* `gpgme_op_keysign`

Right now I have to write XML (create) and shell out (manipulate) to gpg to do that, which I don't don't like.

Additionally I'm also interested in the [[n]ew encryption API to support direct key specification including                                                      
   hidden recipients option and taking keys from a file.](https://git.gnupg.org/cgi-bin/gitweb.cgi?p=gpgme.git;a=blob;f=NEWS;h=9c11df340fd36a98b8656554662d7d05054071a5;hb=HEAD#l77) (I guess that is `gpgme_op_encrypt_ext` and cousins).
